tests/test_spatial_analysis/test_spatial_interaction.py

```diff
@@ -285,6 +360,56 @@ def test_new_ax_has_right_titles(self):
                     error_msg
                 )
 
+    def test_sinlge_stratify_by(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_num",
+            "Neighborhood Enrichment",
+            stratify_by="cluster_str"
+        )
+        unique_cluster_str_values = self.adata.obs["cluster_str"].unique()
+
+        # Get the keys (unique cluster values) from the dictionary
+        keys = list(ax_dict.keys())
+
+        # Assert that we have at least two keys (clusters)
+        self.assertEqual(len(keys), 2)
+
+        # Assert that the axes associated
+        # with the first and second keys are different
+        self.assertNotEqual(ax_dict[keys[0]], ax_dict[keys[1]])
+
+        for value in unique_cluster_str_values:
+            # Expect each unique value as a key in the returned dict
+            self.assertIn(value, ax_dict.keys())
+
+            # Each should be a matplotlib axis object
+            self.assertIsInstance(ax_dict[value], plt.Axes)
+
+    def test_List_stratify_by(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_str",
+            "Neighborhood Enrichment",
+            stratify_by=[
+                "cluster_num",
+                "labels"
+                ]
+            )
+        combined_keys = self.adata.obs[

```

Comment:
@ruiheesi, would you please hardcode the ground truth so that we can check the implemented code is working as expected. Otherwise the test and the function are implementing the exact code, hence the test will always pass without us checking the expected values from the implementation.

Reply:
since this is plotting function, we do not return the adata object nor the zcore and n count with it, hence it is hard to compare "ground truth". And by by ground truth, do we mean the computation from the function itself by simply calling the squidpy functions? Thanks.